crates/red_knot_module_resolver/src/resolver.rs

```diff
@@ -152,54 +166,187 @@ impl RawModuleResolutionSettings {
-        }
+        let site_packages = site_packages
+            .map(|path| Arc::new(ModuleResolutionPathBuf::site_packages(path).unwrap()));
 
         ModuleResolutionSettings {
             target_version,
-            search_paths: OrderedSearchPaths(paths.into_iter().map(Arc::new).collect()),
+            search_path_settings: ValidatedSearchPathSettings {
+                extra_paths,
+                workspace_root,
+                stdlib,
+                site_packages,
+            },
         }
     }
 }
 
-/// A resolved module resolution order as per the [typing spec]
-///
-/// [typing spec]: https://typing.readthedocs.io/en/latest/spec/distributing.html#import-resolution-ordering
-#[derive(Clone, Debug, Default, Eq, PartialEq)]
-pub(crate) struct OrderedSearchPaths(Vec<Arc<ModuleResolutionPathBuf>>);
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub(crate) struct ValidatedSearchPathSettings {
+    extra_paths: Vec<Arc<ModuleResolutionPathBuf>>,
+    workspace_root: Arc<ModuleResolutionPathBuf>,
+    stdlib: Arc<ModuleResolutionPathBuf>,
+    site_packages: Option<Arc<ModuleResolutionPathBuf>>,
+}
 
-impl Deref for OrderedSearchPaths {
-    type Target = [Arc<ModuleResolutionPathBuf>];
+impl ValidatedSearchPathSettings {
+    fn search_paths(&self, db: &dyn Db) -> OrderedSearchPaths {
+        let ValidatedSearchPathSettings {
+            extra_paths,
+            workspace_root,
+            stdlib,
+            site_packages,
+        } = self;
+        let mut search_paths: OrderedSearchPaths = extra_paths
+            .iter()
+            .cloned()
+            .chain([workspace_root, stdlib].into_iter().cloned())
+            .collect();
+        if let Some(site_packages) = site_packages {
+            search_paths.insert(Arc::clone(site_packages));
+            let site_packages = site_packages
+                .as_system_path()
+                .expect("Expected site-packages never to be a VendoredPath!");
+            let Ok(pth_file_iterator) = PthFileIterator::new(db, site_packages) else {
+                return search_paths;
+            };
+
+            // The Python documentation specifies that `.pth` files in `site-packages`
+            // are processed in alphabetical order.
+            // We know that files returned from [`ruff_db::system::MemoryFileSystem::read_directory()`]
+            // will be in alphabetical order already. However, [`ruff_db::system::OsSystem::read_directory()`]
+            // gives no such guarantees. Therefore, collecting and then sorting is necessary.
+            // https://docs.python.org/3/library/site.html#module-site
+            let mut all_pth_files: Vec<PthFile> = pth_file_iterator.collect();
+            all_pth_files.sort_by(|a, b| a.path.cmp(&b.path));
+
+            for pth_file in &all_pth_files {
+                search_paths.extend(pth_file.iter_editable_installations().map(Arc::new));
```

Comment:
These paths aren't normalized and they could be paths relative to `site-packages` rather than absolute paths. We might want to normalize them.